sdk/inc/azure/core/az_json.h

```diff
@@ -278,42 +283,43 @@ typedef struct
 /**
  * @brief Initializes an #az_json_writer which writes JSON text into a buffer.
  *
- * @param[out] json_writer A pointer to an #az_json_writer instance to initialize.
- * @param[in] destination_buffer An #az_span over the byte buffer where the JSON text is to be
- * written.
+ * @param[out] out_json_writer A pointer to an #az_json_writer instance to initialize.
+ * @param destination_buffer An #az_span over the byte buffer where the JSON text is to be written.
  * @param[in] options __[nullable]__ A reference to an #az_json_writer_options
  * structure which defines custom behavior of the #az_json_writer. If `NULL` is passed, the writer
- * will use the default options (i.e. #az_json_writer_options_default()).
+ * will use the default options (i.e. #az_json_writer_options_default).
  *
- * @return An #az_result value indicating the result of the operation:
- *         - #AZ_OK if the az_json_writer is initialized successfully
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK #az_json_writer is initialized successfully.
+ * @retval other Initialization failure.
  */
 AZ_NODISCARD az_result az_json_writer_init(
-    az_json_writer* json_writer,
+    az_json_writer* out_json_writer,
```

Comment:
I don't think we should call this `out_json_writer`.

Revert?
```suggestion
    az_json_writer* json_writer,
```

Reply:
It is an out parameter. Otherwise, there is no output from the function?
It is clearly an out paramter, because in the body, we do not read from it, and only write:
```
_az_PRECONDITION_NOT_NULL(out_json_writer);

  *out_json_writer = (az_json_writer){
    ._internal = {
      .destination_buffer = destination_buffer,
      .allocator_callback = NULL,
      .user_context = NULL,
      .bytes_written = 0,
      .total_bytes_written = 0,
      .need_comma = false,
      .token_kind = AZ_JSON_TOKEN_NONE,
      .bit_stack = { 0 },
      .options = options == NULL ? az_json_writer_options_default() : *options,
    },
  };
  return AZ_OK;
```

If this is not an output parameter, nothing is.

And I know that with out parameters, I can do
```
az_json_writer writer; // no initialization -gets initialized with random bytes from stack
if (az_succeeded(az_json_writer_init(&writer))
{
    writer-> ... // I can be sure that at this point, *writer initialized, and is safe to use, and it didn't matter that I dodn't initialize it with "writer = { 0 };" in the beginning.
}
```